### PR TITLE
Bluetooth: Host: Move BR/EDR files to host/classic

### PR DESCRIPTION
**Pull request by lylezhu2012**
Add subfolder "subsys/bluetooth/host/classic". Move BR/EDR source files to the subfolder.

Move BR/EDR header files from "include/zephyr/bluetooth" to subfolder "include/zephyr/bluetooth/classic".

Move source files of BR/EDR to subfolder "subsys/bluetooth/host/classic/".

Add the path "include/zephyr/bluetooth/classic/" for the header files of BR/EDR.

### CODE REVIEW

File: subsys/bluetooth/host/CMakeLists.txt [hunk 1]
```diff
@@ -3,32 +3,16 @@
 zephyr_library()
 zephyr_library_link_libraries(subsys__bluetooth)
 
+add_subdirectory_ifdef(CONFIG_BT_BREDR classic)
```
Comment (Thalley) [inline subsys/bluetooth/host/CMakeLists.txt:6]: An open question: 
Should we use classic or BREDR everywhere?
Right now, we have the Kconfigs called BREDR, but the directory is called classic. 

So the new directory be `bredr` instead? Or should the Kconfig be renamed to `CONFIG_BT_CLASSIC`? 

That way we would be consistent with BT Mesh and BT Audio

Comment (jori-nordic) [reply subsys/bluetooth/host/CMakeLists.txt:6]: I would also prefer `BT_CLASSIC`.
We can keep the `BREDR` kconfig as deprecated and make it select `CLASSIC`.

File: subsys/bluetooth/host/Kconfig [hunk 2]
```diff
@@ -1028,6 +1028,7 @@ config BT_CONN_DISABLE_SECURITY
 config BT_BREDR
 	bool "Bluetooth BR/EDR support [EXPERIMENTAL]"
 	depends on BT_HCI_HOST
+	select BT_CLASSIC
```
Comment (Thalley) [inline subsys/bluetooth/host/Kconfig:1031]: Given that `BT_BREDR` is considered experimental, can't we just replace the name of the Kconfig? AFAIK we don't do deprecation of experimental things (even if it has been experimental for a long time...) 

If we do a simple rename, then we should put that in the migration guide for 3.7

Comment (lylezhu2012) [reply subsys/bluetooth/host/Kconfig]: It is done. 

Could you guide me how to update the migration guide for 3.7?

Comment (Thalley) [reply subsys/bluetooth/host/Kconfig]: @lylezhu2012 It's basically just adding an entry in https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/migration-guide-3.7.rst (maybe under a new section called Bluetooth Classic next to the Mesh and Audio sections) 

You can see the previous migrations guides at https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/ to get an idea of what to write :)

**Review by Thalley — CHANGES_REQUESTED** [review]
Missing migration guide w.r.t the Kconfig rename

**Comment by lylezhu2012** [comment]
> Missing migration guide w.r.t the Kconfig rename

**Comment by lylezhu2012** [comment]
> Missing migration guide w.r.t the Kconfig rename

Updated migration guide 3.7

File: subsys/bluetooth/host/Kconfig (hunk 2 above)
Comment (lylezhu2012) [reply subsys/bluetooth/host/Kconfig]: Got it. Thanks a lot.

**Review by jori-nordic — DISMISSED** [review]

**Review by alwa-nordic — DISMISSED** [review]

**Review by Thalley — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Comment by jhedberg** [comment]
@lylezhu2012 there seems to be some conflicts that need resolving.

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Pull request merged**